contracts/actions/mstable/MStableDeposit.sol

```diff
@@ -0,0 +1,97 @@
+        address vaultAddress;   // vault contract address for the imAsset (imAssetVault address), unused if stake == false
+        address from;           // address from where to pull the bAsset
+        address to;             // address that will receive the (stake ? imAssetVault : imAsset)
+        uint256 amount;         // amount of bAsset to deposit
+        uint256 minOut;         // minimum amount of mAsset to accept
+        bool stake;             // stake flag
+    }
+
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+        params.bAsset = _parseParamAddr(params.bAsset, _paramMapping[0], _subData, _returnValues);
+        params.mAsset = _parseParamAddr(params.mAsset, _paramMapping[1], _subData, _returnValues);
+        params.saveAddress = _parseParamAddr(params.saveAddress, _paramMapping[2], _subData, _returnValues);
+        params.vaultAddress = _parseParamAddr(params.vaultAddress, _paramMapping[3], _subData, _returnValues);
+        params.from = _parseParamAddr(params.from, _paramMapping[4], _subData, _returnValues);
+        params.to = _parseParamAddr(params.to, _paramMapping[5], _subData, _returnValues);
+        params.amount = _parseParamUint(params.amount, _paramMapping[6], _subData, _returnValues);
+        params.minOut = _parseParamUint(params.minOut, _paramMapping[7], _subData, _returnValues);
+        params.stake = _parseParamUint(params.stake ? 1 : 0, _paramMapping[8], _subData, _returnValues) == 0 ? false : true;
+        
+        uint256 credits = _mStableDeposit(params);
+        return bytes32(credits);
+    }
+
+    function executeActionDirect(bytes[] memory _callData) public payable override {
+        Params memory params = parseInputs(_callData);
+        _mStableDeposit(params);
+    }
+
+    /// @notice Action that deposits the base asset into the mStable Savings Contract and optionally stakes the received imAsset into the Savings Vault
+    function _mStableDeposit(Params memory _params) internal returns (uint256 credits) {
+        // _params.to = 0 will revert
+        // _params.amount = 0 will revert
+        if (_params.amount == type(uint256).max) {
+            _params.amount = _params.bAsset.getBalance(_params.from);
+        }
+        _params.bAsset.pullTokensIfNeeded(_params.from, _params.amount);
+        _params.bAsset.approveToken(_params.mAsset, _params.amount);
+
+        uint256 mAssetsMinted = ImAsset(_params.mAsset).mint(_params.bAsset, _params.amount, _params.minOut, address(this));
+
+        _params.mAsset.approveToken(_params.saveAddress, mAssetsMinted);
+        if (_params.stake) {
+            credits = ISavingsContractV2(_params.saveAddress).depositSavings(mAssetsMinted, address(this));
```

Comment:
credits = ISavingsContractV2(_params.saveAddress).depositSavings(mAssetsMinted, stake ? address(this) : _params.to);

Can pull this outside the if statement as the same operation is done in both cases just a diff. input.

Reply:
I think readability is better this way.